en/fpv-openipc-aio-internet-forwarding.md

```diff
@@ -0,0 +1,132 @@
+## AIO Mario firmware update over usb via internet forwarding
+This method will work on Linux due to specific network commands. The key idea is to use the local PC as a network bridge betwen AIO board ethernet interface and your local internet interface.
+
+## Steps
+1. Connect AIO Mario to USB
+2. Identify your PC address. On your local PC run:
+```bash
+ip -c a
+...
+3: enx00e099fead02: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
+    link/ether 00:e0:99:fe:ad:02 brd ff:ff:ff:ff:ff:ff
+    inet 192.168.1.11/24 brd 192.168.1.255 scope global noprefixroute enx00e099fead02
+       valid_lft forever preferred_lft forever
+    inet6 fe80::de74:78fc:9af0:f031/64 scope link noprefixroute 
+       valid_lft forever preferred_lft forever
+```
+Here: `192.168.1.11` is my local PC address on AIO ethernet interface.
+
+3. Connect to AIO Mario over ssh. On PC:
+```bash
+ssh root@192.168.1.10
+```
+4. List AIO routes. On AIO run:
+```bash
+root@openipc-ssc338q:~# route -n
+Kernel IP routing table
+Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
+192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 eth0
+```
+5. We need to add a route to our local PC, so traffic could get route over. On AIO run:
+```bash
+root@openipc-ssc338q:~# route add default gw 192.168.1.11 eth0
+```
+Here: `192.168.1.11` is the local PC address.
+
+6. Verify that route was added correctly:
+```bash
+root@openipc-ssc338q:~# route -n
+Kernel IP routing table
+Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
+0.0.0.0         192.168.1.11    0.0.0.0         UG    0      0        0 eth0
+192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 eth0
+```
+
+7. On the local PC we need to forward all the traffic from `enx00e099fead02` AIO Mario ethernet interface to our local connection.
+This can be `wlp2s0` for wifi. To check the interface name, on the local PC run again:
+```bash
+ip -c a
+2: wlp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
+    link/ether f8:34:41:af:55:37 brd ff:ff:ff:ff:ff:ff
+    inet 192.168.1.8/24 brd 192.168.1.255 scope global noprefixroute wlp2s0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::8d6a:f2f6:c850:cf9d/64 scope link noprefixroute 
+       valid_lft forever preferred_lft forever
+```
+Here: `wlp2s0` is my internet connection, it could be also `eth0` on other setups.
+
+8. Traffic forwarding. Create a bash script `~/forward.sh`:
+```bash
+#!/bin/bash
+
+# Get the interface name of the wireless card by stdin
+# $1 is the interface name of the AIO card
+
+USB_AIO=$1
+echo $1
+
+sudo iptables -A FORWARD -i $USB_AIO -o wlp2s0 -j ACCEPT
```

Comment:
Maybe take the internet interface as parameter as well instead of hardcoding it?

Reply:
Script updated accordingly